modules/plugins/wfly-10/src/main/java/org/rhq/modules/plugins/wildfly10/NamingBindingComponent.java

```diff
@@ -0,0 +1,107 @@
+import org.rhq.modules.plugins.wildfly10.json.CompositeOperation;
+import org.rhq.core.domain.configuration.Property;
+import org.rhq.modules.plugins.wildfly10.json.WriteAttribute;
+
+/**
+ * Component for the naming subsystem
+ * @author Ruben Vargas
+ */
+public class NamingBindingComponent extends BaseComponent<BaseComponent<?>> {
+
+    private static final String ENVIRONMENT = "environment";
+
+    class BindingConfigurationWriteDelegate extends ConfigurationWriteDelegate {
+
+
+        public BindingConfigurationWriteDelegate(ConfigurationDefinition configDef, ASConnection connection, Address
+                address) {
+            super(configDef.copy(), connection, address);
+            this.configurationDefinition.getPropertyDefinitions().remove(ENVIRONMENT);
+        }
+
+        @Override
+        protected CompositeOperation updateGenerateOperationFromProperties(Configuration conf, Address address) {
+            CompositeOperation cop = super.updateGenerateOperationFromProperties(conf, address);
+            /* Update special enviroment var */
+            PropertyList prop = (PropertyList)conf.get(ENVIRONMENT);
+            List<Object> values = new ArrayList<Object>();
+            List<Property> embeddedProps = prop.getList();
+            Map<String, Object> results = new HashMap<String, Object>();
+            for (Property inner : embeddedProps) {
+                PropertyMap propMap = (PropertyMap) inner;
+                PropertySimple propName = ((PropertySimple) propMap.get("name"));
+                PropertySimple propValue = ((PropertySimple) propMap.get("value"));
+                results.put(propName.getStringValue(), propValue.getStringValue());
+            }
+            WriteAttribute writeAttribute = new WriteAttribute(address, ENVIRONMENT, results);
+            cop.addStep(writeAttribute);
+            return cop;
+        }
+    }
+
+    @Override
+    public void updateResourceConfiguration(ConfigurationUpdateReport report) {
+        ConfigurationDefinition configDef = context.getResourceType().getResourceConfigurationDefinition().copy();
+        ConfigurationWriteDelegate delegate = new BindingConfigurationWriteDelegate(configDef, getASConnection(), address);
+        delegate.updateResourceConfiguration(report);
+    }
+
+    @Override
+    public Configuration loadResourceConfiguration() throws Exception {
+        ConfigurationDefinition configDef = context.getResourceType().getResourceConfigurationDefinition().copy();
+        /* Get and remove enviroment property */
+        configDef.getPropertyDefinitions().remove(ENVIRONMENT);
+
+        /* Load other properties*/
+        ConfigurationLoadDelegate delegate = new ConfigurationLoadDelegate(configDef, getASConnection(), address,
+                includeRuntime);
+        Configuration configuration = delegate.loadResourceConfiguration();
+
+        Operation op = new Operation("whoami", getAddress());
```

Comment:
Why? Why not just check the result of the real operation if it fails or not?

Reply:
I just take it from existing code :) but yes, we can check just for the result, I can change it.

Reply:
Not sure if this is what you meant:

```java
        //:whoami might fail host controller resources, in that case use :read-resource operation
        //which is slower due to larger content returned but more reliable since every resource has it.

```
https://github.com/rhq-project/rhq/blob/master/modules/plugins/wfly-10/src/main/java/org/rhq/modules/plugins/wildfly10/BaseComponent.java#L369-L370

Reply:
As I saw this is just for know if the server needs to be restarted or  need to be reloaded after configuration changes, so the "real" operation here is `whoami`, which is used just for check for those two things (reload and restart flags), the `resource-reading` op is a fallback in cases where `whoami` fails.